aquadoggo/src/graphql/schema.rs

```diff
@@ -39,3 +47,143 @@ pub fn build_root_schema(store: SqlStorage, tx: ServiceSender) -> RootSchema {
         .data(tx)
         .finish()
 }
+
+/// Location of the temporary file containing all currently known schemas.
+pub const TEMP_FILE_PATH: &'static str = "./.schemas.tmp.json";
+
+/// Returns GraphQL API schema for p2panda node with a little trick to make dynamic schemas work.
+///
+/// The `async_graphql` crate we're using in this project does only provide methods to generate
+/// GraphQL schemas statically. Ideally we would like to query our database for currently known
+/// p2panda schemas and accordingly update the GraphQL schema whenever necessary but we don't have
+/// static access to the database when building `async_graphql` types.
+///
+/// With this little workaround we are still able to make it work! We load the p2panda schemas from
+/// the database and serialise them into a JSON file. When `async_graphql` builds the GraphQL
+/// schema we can load this file statically to build the schemas dynamically based on the file's
+/// content.
+async fn build_schema_with_workaround(shared: GraphQLSharedData) -> RootSchema {
+    // Store temporary JSON file with all serialised application schemas from database.
+    //
+    // @TODO: We could try to still access the database by creating a static interface to it?
```

Comment:
:point_up: I'd like to explore this for a short while, just to see how far we can get.